### PR TITLE
Rename the classical diff_method to backprop

### PR DESCRIPTION
**Pull request by josh146**
**Description of the Change:** Renames `diff_method="classical"` to `"backprop"` 

**Benefits:** Better describes the differentiation method than "classical". Note that I have not updated the changelog, as this will cause a non-trivial clash with #634; I will update it there once merged in.

**Possible Drawbacks:** n/a

**Related GitHub Issues:** n/a

### CODE REVIEW

File: tests/beta/test_default_tensor_tf.py [hunk 1]
```diff
@@ -774,7 +774,7 @@ def test_error_classical_diff_autograd(self, tol):
         """Tests that an error is raised if for the classical differentiation
         method when using the autograd interface"""
         interface = "autograd"
-        diff_method = "classical"
+        diff_method = "backprop"
```
Comment (antalszava) [inline tests/beta/test_default_tensor_tf.py:777]: Two more occurrences on lines [770](https://github.com/XanaduAI/pennylane/blob/178b772435e1f22e19afc8aa828d9221d20ad719/tests/beta/test_default_tensor_tf.py#L770), [792](https://github.com/XanaduAI/pennylane/blob/178b772435e1f22e19afc8aa828d9221d20ad719/tests/beta/test_default_tensor_tf.py#L792) too at the end of the error message

Comment (josh146) [reply tests/beta/test_default_tensor_tf.py:777]: Good catch my `sed` skills failed me!

Comment (josh146) [reply tests/beta/test_default_tensor_tf.py:777]: Oh you know why? I only searched for `"classical"`, not `'classical'` 🤦‍♂️

Comment (co9olguy) [reply tests/beta/test_default_tensor_tf.py:777]: weird, I don't see this

Comment (co9olguy) [reply tests/beta/test_default_tensor_tf.py:777]: ahh, I see it now, nevermind :facepalm:

**Pull request merged**